src/hooks/useKeyboardNavigation.ts

```diff
@@ -0,0 +1,54 @@
+
+import { useState } from "react";
+import useEventListener from "./useEventListener";
+
+const KEY_HANDLERS = {
+    UP: "onUp",
+    DOWN: "onDown",
+    LEFT: "onLeft",
+    RIGHT: "onRight",
+    ENTER: "onEnter",
+    ESCAPE: "onEscape",
+};
+
+const KEY_CODES = {
+    UP: 38,
+    DOWN: 40,
+    LEFT: 37,
+    RIGHT: 39,
+    ENTER: 13,
+    ESCAPE: 27,
+};
+
+export type KeyBoardEvents = {
+    onUp?: Function,
+    onDown?: Function,
+    onLeft?: Function,
+    onRight?: Function,
+}
+
+export default (keyBoardEvents: KeyBoardEvents) => {
+    const [ref, setRef] = useState(null);
```

Comment:
Would useState<HTMLElement | null>(null); prevent setting other random values as the ref?